include/vcpkg/base/git.h

```diff
@@ -38,16 +38,50 @@ namespace vcpkg
         std::string old_path;
     };
 
+    /* ===== Git command abstractions  =====*/
+    // run git status on a repository, optionaly a specific subpath can be queried
+    ExpectedL<std::vector<GitStatusLine>> git_status(const GitConfig& config, StringView path = {});
```

Comment:
I'm not sure how this is used throughout the code, however it seems to me that the exe path will generally take on only a single value. However, the other settings (git_dir, git_work_tree) will potentially change on a call-by-call basis.

This makes me think that perhaps it would be better to structure this as:
```cpp
struct GitConfig {
    Optional<Path> dir, work_tree;
};
struct GitTool {
    ExpectedL<std::vector<GitStatusLine>> status(const GitConfig& config, StringView path = {});
};
```
Because these commands also are likely to have many (possibly optional) arguments, it might make sense to use an argument struct like:
```cpp
struct FooArgs {
    Optional<Path> cwd;
    Optional<Path> work_tree;
    StringView subpath;
};

int foo(const FooArgs& args);

int main(){ 
    foo({.cwd = "/cwd", /* work_tree left as nullopt */, .subpath="versions/baseline.json"});
}
```

Reply:
This seems like a separate / different improvement than sucking stuff out of vcpkgpaths tho